docs/development-guide.md

```diff
@@ -0,0 +1,192 @@
+Origin (DCO). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution. See the [DCO](DCO) file for details.
+
+Contributors sign-off that they adhere to these requirements by adding a
+Signed-off-by line to commit messages. For example:
+
+```
+subsystem: This is my commit message
+
+More details on what this commit does
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+If you have already made a commit and forgot to include the sign-off, you can
+amend your last commit to add the sign-off with the following command, which
+can then be force pushed.
+
+```console
+git commit --amend -s
+```
+
+We use a [DCO bot](https://github.com/apps/dco) to enforce the DCO on each pull
+request and branch commits.
+
+#### Commit Messages
+
+We follow a rough convention for commit messages that is designed to answer two
+questions: what changed and why? The subject line should feature the what and
+the body of the commit should describe the why.
+
+```
+controller: fix bug in configmap
+
+fix clusterID bug in the configmap where the clusterID is
+not set to the expected value.
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+
+The format can be described more formally as follows:
+
+```
+<component>: <subject of the change>
+<BLANK LINE>
+<paragraph(s) with reason/description>
+<BLANK LINE>
+<signed-off-by>
+```
+
+The first line is the subject and should be no longer than 70 characters, the
+second line is always blank, and other lines should be wrapped at 80 characters.
+This allows the message to be easier to read on GitHub as well as in various
+git tools.
+
+Here is a short guide on how to work on a new patch.  In this example, we will
+work on a patch called *hellopatch*:
+
+```console
+git checkout main
+git pull
+git checkout -b hellopatch
+```
+
+Do your work here and commit.
+
+Run the test suite, which includes linting checks, static code check, and unit
+tests:
+
+```console
+make test
+```
+
+Once you are ready to push, you will type the following:
+
+```console
+git push fork hellopatch
+```
+
+**Creating A Pull Request:**
+When you are satisfied with your changes, you will then need to go to your repo
+in GitHub.com and create a pull request for your branch. Automated tests will
+be run against the pull request. Your pull request will be reviewed and merged.
+
+If you are planning on making a large set of changes or a major architectural
+change it is often desirable to first build a consensus in an issue discussion
+and/or create an initial design doc PR. Once the design has been agreed upon
+one or more PRs implementing the plan can be made.
+
+A few labels interact with automation around the pull requests:
+
+* ready-to-merge: This PR is ready to be merged and it doesn't need second review
+* DNM: DO NOT MERGE (Mergify will not merge this PR)
+* ok-to-test: PR is ready for e2e testing.
```

Comment:
I am not sure we have these integrations, and using labels with these specific names (I think we are using different labels today)

Reply:
we dont have anything today. once the PR is merged, i will create the required labels.

Reply:
Ok in that case why DNM and not the label we use everywhere else: `don-not-merge`